ethcore/light/src/client/service.rs

```diff
@@ -56,6 +60,8 @@ pub struct Service {
 impl Service {
 	/// Start the service: initialize I/O workers and client itself.
 	pub fn start(config: ClientConfig, spec: &Spec, path: &Path) -> Result<Self, Error> {
+		let cache = Arc::new(Mutex::new(Cache::new(Default::default(), Duration::hours(6))));
```

Comment:
This should be given a cache, which should be the same one given to `OnDemand` and RPC in `parity/run.rs`